voidcrew/code/controllers/subsystem/overmap.dm

```diff
@@ -198,54 +198,20 @@ SUBSYSTEM_DEF(overmap)
   */
-/datum/controller/subsystem/overmap/proc/spawn_dynamic_encounter(planet_type, ruin = TRUE, ignore_cooldown = FALSE, datum/map_template/ruin/ruin_type)
+/datum/controller/subsystem/overmap/proc/spawn_dynamic_encounter(datum/overmap/planet/planet_type, ruin = TRUE, ignore_cooldown = FALSE, datum/map_template/ruin/ruin_type)
 	var/list/ruin_list
 	var/datum/map_generator/mapgen
 	var/area/target_area
 	var/turf/surface = /turf/open/space
 	var/datum/weather_controller/weather_controller_type
-	if(planet_type)
-		switch(planet_type)
-			if(DYNAMIC_WORLD_LAVA)
-				ruin_list = SSmapping.lava_ruins_templates
-				mapgen = new /datum/map_generator/cave_generator/lavaland
-				target_area = /area/overmap_encounter/planetoid/lava
-				surface = /turf/open/floor/plating/asteroid/basalt/lava_land_surface
-				weather_controller_type = /datum/weather_controller/lavaland
-			if(DYNAMIC_WORLD_ICE)
-				ruin_list = SSmapping.ice_ruins_templates
-				mapgen = new /datum/map_generator/cave_generator/icemoon
-				target_area = /area/overmap_encounter/planetoid/ice
-				surface = /turf/open/floor/plating/asteroid/snow/icemoon
-				weather_controller_type = /datum/weather_controller/snow_planet
-			if(DYNAMIC_WORLD_SAND)
-				ruin_list = SSmapping.sand_ruins_templates
-				mapgen = new /datum/map_generator/cave_generator/whitesands
-				target_area = /area/overmap_encounter/planetoid/sand
-				surface = /turf/open/floor/plating/asteroid/whitesands
-				weather_controller_type = /datum/weather_controller/desert
-			if(DYNAMIC_WORLD_JUNGLE)
-				ruin_list = SSmapping.jungle_ruins_templates
-				mapgen = new /datum/map_generator/jungle_generator
-				target_area = /area/overmap_encounter/planetoid/jungle
-				surface = /turf/open/floor/plating/dirt/jungle
-				weather_controller_type = /datum/weather_controller/lush
-			if(DYNAMIC_WORLD_ASTEROID)
-				ruin_list = null
-				mapgen = new /datum/map_generator/cave_generator/asteroid
-			if(DYNAMIC_WORLD_ROCKPLANET)
-				ruin_list = SSmapping.rock_ruins_templates
-				mapgen = new /datum/map_generator/cave_generator/rockplanet
-				target_area = /area/overmap_encounter/planetoid/rockplanet
-				surface = /turf/open/floor/plating/asteroid
-				weather_controller_type = /datum/weather_controller/chlorine //let's go??
-			if(DYNAMIC_WORLD_REEBE)
-				ruin_list = SSmapping.yellow_ruins_templates
-				mapgen = new /datum/map_generator/cave_generator/reebe
-				target_area = /area/overmap_encounter/planetoid/reebe
-				surface = /turf/open/chasm/reebe_void
-			if(DYNAMIC_WORLD_SPACERUIN)
-				ruin_list = SSmapping.space_ruins_templates
+	if(!isnull(planet_type))
+		planet_type = new planet_type
+		ruin_list = planet_type.ruin_list
+		mapgen = new planet_type.mapgen
+		target_area = planet_type.target_area
+		surface = planet_type.surface
+		weather_controller_type = planet_type.weather_controller_type
+		qdel(planet_type)
```

Comment:
we really must qdel datum? This don't create runtime?

Reply:
I have to make a new one because `ruins_list` is only filled on the datum's `New`. Because SSmapping can't be accessed from assignment. Qdelling is fine because i created a new instnace of it right above

Reply:
As i see probabilities already list of generated datums, that already must have filled ruins_list.
On the other hand, if make a random generation of fullness of resources, a random atmosphere and weather.
And you can store this vars in datum.